### PR TITLE
feat(mobile): render login fields/buttons based on server configuration

### PR DESCRIPTION
**Pull request by ddshd**
Issue #3337, Discussion #2206 

Shows/hides email/password fields based on config on the server
Shows/hides oauth buttons based on config on the server
Shows error message same as web interface is both are disabled

Only Password
Both Enabled
Only OAuth
Nothing Enabled

### CODE REVIEW

File: mobile/lib/modules/login/ui/login_form.dart [hunk 1]
```diff
@@ -317,6 +326,10 @@ class LoginForm extends HookConsumerWidget {
                       ],
                     ],
                   ),
+            if (!isOauthEnable.value && !isPasswordLoginEnable.value)
+              const Center(
+                child: Text("Login has been disabled."),
```
Comment (ddshd) [inline mobile/lib/modules/login/ui/login_form.dart:331]: How to support translation? Just add a field to the English json?

**Comment by jrasm91** [comment]
Can you attach some screenshots to the PR description that show the updated flow?

File: mobile/lib/modules/login/ui/login_form.dart (hunk 1 above)
Comment (alextran1502) [reply mobile/lib/modules/login/ui/login_form.dart:331]: Correct, and the end of the Text it should have a `tr()` method. You can reference other files for example

**Review by alextran1502 — APPROVED** [review]

**Comment by alextran1502** [comment]
@ddshd Can you allow the maintainer to push to your branch? 

Here is how you can get the key rendered correctly 
image

**Comment by ddshd** [comment]
> @ddshd Can you allow the maintainer to push to your branch?
> 
> Here is how you can get the key rendered correctly image

Flutter threw an error when i did that for some reason but 

Enabled access for changes

**Pull request merged**